### ISSUE TITLE
startNew throws EPROTO error locally

### ISSUE THREAD

**Issue by kashimiz** [Azure/azure-functions-durable-js#28]
When running locally, making calls to `DurableOrchestrationClient.startNew` to start an orchestrator function results in an EPROTO error being thrown and the orchestrator not starting.

This is due to Azure/azure-functions-host#2024, which causes the DurableOrchestrationClient class to use the wrong webhook URLs. The workaround is to set local environment variable `WEBSITE_HOSTNAME` to `localhost:<port>`, ex. `localhost:7071`.

**Comment by sckelly** [comment]
Hey Kashimiz, thanks for flagging this.
I'm trying to set up the starter project as found here:
https://docs.microsoft.com/en-us/azure/azure-functions/durable/quickstart-js-vscode

I'm getting the EPROTO error that you have mentioned here I believe, it relates to SSL wrong version number? (Just want to be sure I'm not mismatching issues). HttpStart won't kick off.
I've tried setting the Local env variable as you've mentioned in your workaround, but still no joy.

Is there any more info on this?
Thanks and all the best.

**Comment by kashimiz** [comment]
Thanks for reporting you ran into this error, @sckelly. It's interesting that you're still running into that error.. Could you confirm a couple of things to rule out some possible causes?

1) What version of the Durable Functions extension do you have installed? Should be 1.7.0 in your app's `extensions.csproj` file.
2) Did you restart your editor after setting the local environment variable?
3) Are you able to start your function app and call HttpStart? (Just execute it, not necessarily return a good response.) If so, can you:
   - set a breakpoint in your HttpStart function just after your function declaration
   - call HttpStart
   - let me know the value of the following variable: `context.bindings.starter.creationUrls.createNewInstancePostUri`

**Comment by sckelly** [comment]
I ended up fixing the error with your work-around mentioned above! I wasn't paying attention properly so that's my bad. I think everything is up to date: locally I can run OrchestratorFunction to get the output I want.

However when deploying to Azure I'm running into an error where I am told URL is not defined..
I know here really isn't the place for that. I shall do some googling and see what I can see!

Thanks again for the help and flagging this, it's good to know what's going on! :smile_cat:

**Comment by sckelly** [comment]
Though trying to fix the deployment now, I get another issue in regards to URL not defined in the DurableOrchestrationClient.. (this may be something simple I'm missing like an env param)

**Comment by kashimiz** [comment]
Hi @sckelly, my apologies for the delayed response. It sounds like you've still got an older version of `durable-functions` (< 1.1.2) installed if you're running into the URL not defined issue; were you able to get this resolved?

I have a fix out for this issue in [this PR](https://github.com/Azure/azure-functions-durable-js/pull/37).

**Comment by simonagren** [comment]
Hey @kashimiz 
I have that Eproto error.
I tried to set the WEBSITE_HOSTNAME in local.settings.json, but then I got an error saying that key has already been set.
I debug from VS Code and should have all the right stuff installed. I also run the Emulator and these are the local variables I use:

```json
"AzureWebJobsStorage": "UseDevelopmentStorage=true",
    "AzureWebJobsSecretStorageType": "files",
    "FUNCTIONS_WORKER_RUNTIME": "node",
```

Any ideas?

**Comment by kashimiz** [comment]
Hi @simonagren, try setting WEBSITE_HOSTNAME directly in your system environment variables.

**Comment by simonagren** [comment]
@kashimiz Genius :D Thanks!

**Comment by kashimiz** [comment]
This has been fixed in the durable-functions 1.1.3 release.
